python/pynessie/decorators.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Top-level decorators."""
+import functools
+import sys
+from typing import Any, Callable
+
+import click
+from marshmallow import ValidationError
+
+from .cli_common_context import ContextObject
+from .error import NessieCliError, NessieException
+
+
+_REMEDIES = {
+    "Cannot create an unassigned tag reference": "set a valid reference on which to create this tag. eg `nessie tag tag_name main`"
```

Comment:
nit: I do not think this works anymore. I could not find a place where the server might use this particular error message.

Reply:
@rymurr : Would it make sense to add fix suggestions to the original server-generated error message? This way all clients would benefit.

Reply:
> @rymurr : Would it make sense to add fix suggestions to the original server-generated error message? This way all clients would benefit.

I think this makes sense, so the client can use that if it exists in the output exception JSON. 

P/S: Thanks for checking @dimas-b ! Indeed it seems to be not used, I will just remove it